crates/ra_lsp_server/src/path_map.rs

```diff
@@ -33,7 +33,7 @@ impl PathMap {
         let file_id = self
             .path2id
             .get(path.as_path())
-            .map(|&id| id)
+            .cloned()
             .unwrap_or_else(|| {
```

Comment:
I’d rather not used `cloned` here: it’s important to see that this is a cheap copy rather then a potentially expensive clone.  Everything else lgtm